src/wp-includes/rest-api/search/class-wp-rest-post-search-handler.php

```diff
@@ -58,12 +58,13 @@ public function search_items( WP_REST_Request $request ) {
 		}
 
 		$query_args = array(
-			'post_type'           => $post_types,
-			'post_status'         => 'publish',
-			'paged'               => (int) $request['page'],
-			'posts_per_page'      => (int) $request['per_page'],
-			'ignore_sticky_posts' => true,
-			'fields'              => 'ids',
+			'post_type'              => $post_types,
+			'post_status'            => 'publish',
+			'paged'                  => (int) $request['page'],
+			'posts_per_page'         => (int) $request['per_page'],
+			'ignore_sticky_posts'    => true,
+			'update_post_term_cache' => false,
+			'update_post_meta_cache' => false,
```

Comment:
```suggestion
			'update_post_term_cache' => isset( $_GET['_embed'] ),
			'update_post_meta_cache' => isset( $_GET['_embed'] ),
```

I suggest this be conditional on whether or not posts are embedded in the response.

It's relatively simple for a developer to include either a term or some post meta in an endpoint via `register_rest_field()` so this could end up increasing the number of queries on a site that does so.

If you are eager, you may be able to check if any fields are registered via the `$wp_rest_additional_fields` global.

Reply:
I don't quite understand this, do you mind elaborating? Thanks!

Reply:
Oh, I see, in the response when _embed is present wp: terms are retrieved. But I don't see the use of meta, how is it depending from _embed ?

Reply:
Neither the item or links load meta or term here

https://github.com/WordPress/wordpress-develop/blob/15deb80e504a6ede868bedc1e64e448536d88771/src/wp-includes/rest-api/search/class-wp-rest-post-search-handler.php#L104-L164

Why do we need to prime these?

Reply:
The likelyhood that this request would be embedded seems small here. Searches are done by users depending on request. When would this be embedded and it what context.

Reply:
The search endpoint is actually a common enough embed target IMO. It essentially our recommended way for getting posts across multiple post types in one request and embedding allows you to not need to make a subsequent query to get broader details for each item.

That being said, I would not be a fan of us breaking encapsulation inside the controller here by looking at the global `$_GET['embed']` value. Not only for purity take, but it can simply be incorrect because that global may not correspond to this actual request object being processed by the server.

That all being said, I think we could punt support for making it conditional on embedding for now and open up another ticket for making the `WP_REST_Request` object be able to carry additional metadata about the request like the requested embed items.

Reply:
> That all being said, I think we could punt support for making it conditional on embedding for now and open up another ticket for making the `WP_REST_Request` object be able to carry additional metadata about the request like the requested embed items.

In the mean time is it worth always warming the data? I am in two minds.

Thanks both for answering Manu's question while I was on PTO.

Reply:
I'm fine either way. In Gutenberg when we use the search endpoint we don't `embed`, so from a Core perspective I don't think we need it.

https://github.com/WordPress/gutenberg/blob/bc7715732c5371b08ba18a4ba379d0aa69319768/packages/core-data/src/fetch/__experimental-fetch-link-suggestions.js

Reply:
> That being said, I would not be a fan of us breaking encapsulation inside the controller here by looking at the global `$_GET['embed']` value. Not only for purity take, but it can simply be incorrect because that global may not correspond to this actual request object being processed by the server.

How about `isset( $request['_embed']`, is that what I meant? I think that covers POST and other request types too, correct?

Reply:
That would be more accurate. But I still think it'd be better to add that as metadata to the `WP_REST_Request` object so that we have a proper API from the start. Whether embedding happens is really a function of how `WP_REST_Server::response_to_data` is called, so it still isn't a guarantee.

Reply:
Thanks, for now let's warm the full cache always (terms and meta) so it caches embeds and customised endpoints. 

This PR is still a huge performance boost for the endpoint and we can consider the last bit of optimisation as a followup.

Reply:
Sounds good!

Reply:
Hello guys, thanks for the follow-ups, if understood correctly, i should warm up always meta and terms:
'update_post_term_cache' => true,
'update_post_meta_cache' => true,
and P.R is good to go? is that right?

Reply:
I am yet to understand why term and meta caches need to be primed?

Reply:
> I am yet to understand why term and meta caches need to be primed?

If the embed option is used for the search endpoint `http://wp-dev.local/wp-json/wp/v2/search/?search=some-term&_embed` then each found post's REST endpoint is included in the result. 

As it's relatively trivial to add custom data from either meta or terms to the post endpoint, it seems wise to warm their caches too. 

As the embed includes `post.content.rendered` it's also possible any sites using server side rendered blocks will hit the post meta. I'm thinking about devs using something along the lines of ACF blocks as an example.

Reply:
> and P.R is good to go? is that right?

Yes, that's correct, it should be good to go after that.

Reply:
PR updated